x/rollapp/types/liveness.go

```diff
@@ -0,0 +1,67 @@
+package types
+
+import (
+	"bytes"
+	"encoding/binary"
+)
+
+var (
+	LivenessEventQueueKeyPrefix = []byte("LivenessEventQueue")
+	LivenessEventQueueSlash     = []byte("s")
+	LivenessEventQueueJail      = []byte("j")
+)
+
+func LivenessEventQueueKey(e LivenessEvent) []byte {
+	v := LivenessEventQueueSlash
+	if e.IsJail {
+		v = LivenessEventQueueJail
+	}
+	return createLivenessEventQueueKey(&e.HubHeight, v, &e.RollappId)
+}
+
+func LivenessEventQueueIterKey(height *int64) []byte {
```

Comment:
why it needs to be a pointer? in what cases u'll have a LivenessEvent with no height?
seems rednundent and complex

Reply:
the events always have a height, this is for iterating
so if you don't specify a height you will iterate over all